### PR TITLE
Advanced Trip Form Panel

### PR DESCRIPTION
**Pull request by amy-corson-ibigroup**
**Description:**

- Creates new Advanced trip form
- Creates "more options" button that opens advanced trip form
- Animates advanced trip form in and out
- Creates `prefersReducedMotion` util 

**PR Checklist:**
- [x] Does the code follow accessibility standards (WCAG 2.1 AA Compliant)?
- [ ] Are all languages supported (Internationalization/Localization)? - French needed
- [x] Are appropriate Typescript types implemented?

| Before | After |
|--------|-------|
|image|image|

### CODE REVIEW

**Review by josh-willis-arcadis — APPROVED** [review]
A great start to build from.

File: lib/components/util/prefersReducedMotion.tsx [hunk 1]
```diff
@@ -0,0 +1,9 @@
+/**
+ * Identifies whether the user's machine has "reduced motion" enabled
+ *  in their local settings. If reduced motion is on, the app should
+ * show as few animations & transitions as possible.
+ * @returns boolean reflecting whether user prefers reduced motion
+ */
+export const prefersReducedMotion = window.matchMedia(
+  '(prefers-reduced-motion: reduce)'
```
Comment (daniel-heppner-ibigroup) [inline lib/components/util/prefersReducedMotion.tsx:8]: It is also possible to check for this in the CSS directly https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-reduced-motion

File: lib/components/form/advanced-settings-panel.tsx [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+          <h1 className="header-text">{headerText}</h1>
+          <CloseButton
+            aria-label={closeButtonText}
+            onClick={closePanel}
+            title={closeButtonText}
+          >
+            <Close size={22} />
+          </CloseButton>
+        </HeaderContainer>
+        <PageTitle title={headerText} />
```
Comment (daniel-heppner-ibigroup) [inline lib/components/form/advanced-settings-panel.tsx:76]: Can this go somewhere else? Feels random here

File: lib/components/app/batch-routing-panel.tsx [hunk 3]
```diff
@@ -33,6 +49,22 @@ class BatchRoutingPanel extends Component<Props> {
     this.setState({ planTripClicked: true })
   }
 
+  handleOpenAdvanceSettings = () => {
+    this.setState({ showAdvancedModeSettings: true })
+    // Allow Advanced Settings panel to finish animation before removing form from DOM
+    setTimeout(() => {
+      this.setState({ fade: true })
+    }, PANEL_ANIMATION_TIMING)
```
[the remark this answers is not in the document]
Comment (amy-corson-ibigroup) [reply lib/components/app/batch-routing-panel.tsx:76]: I can look into it! Might be cleaner especially if we want to incorporate more transitions app-wide

[the remark this answers is not in the document]
Comment (amy-corson-ibigroup) [reply lib/components/app/batch-routing-panel.tsx]: that was so annoying [fc07097](https://github.com/opentripplanner/otp-react-redux/pull/1230/commits/fc070976ca9c2d95878d0bf19b3755a99c75bb2f)

File: lib/components/util/prefersReducedMotion.tsx (hunk 1 above)
Comment (amy-corson-ibigroup) [reply lib/components/util/prefersReducedMotion.tsx:8]: Yes but since we're using `setTimeout`, I think we need it here too!

**Review by daniel-heppner-ibigroup — APPROVED** [review]
Really nice work!!

File: lib/components/form/advanced-settings-panel.tsx [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+}): JSX.Element => {
+  const intl = useIntl()
+  const closeButtonText = intl.formatMessage({
+    id: 'components.BatchSearchScreen.closeAdvancedPreferences'
+  })
+  const headerText = intl.formatMessage({
+    id: 'components.BatchSearchScreen.advancedHeader'
+  })
+
+  const closePanel = () => {
```
[the remark this answers is not in the document]
Comment (amy-corson-ibigroup) [reply lib/components/form/advanced-settings-panel.tsx:100]: You're right!! A holdover from before `CSSTransition`

**Pull request merged**